File: monsoon3.0/src/main/java/wtf/monsoon/impl/event/EventBlockCollide.java
package wtf.monsoon.impl.event;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import net.minecraft.block.Block;
import net.minecraft.util.AxisAlignedBB;
import wtf.monsoon.api.event.Event;

@Getter
@Setter
@AllArgsConstructor
public final class EventBlockCollide extends Event {
    private AxisAlignedBB collisionBoundingBox;
    private Block block;
    private int x, y, z;
}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/event/EventBlockCollide.kt
package wtf.monsoon.client.event

import me.bush.eventbuskotlin.Event
import net.minecraft.block.Block
import net.minecraft.client.gui.ScaledResolution
import net.minecraft.util.AxisAlignedBB

data class EventBlockCollide(var collisionBoundingBox: AxisAlignedBB?, var block: Block?, var x: Double?, var y: Double?, var z: Double?) : Event() {
    override val cancellable: Boolean = true
}